From 56bbaa97d3f3e232e554891d1809e3beceec20e6 Mon Sep 17 00:00:00 2001
From: Niko Tyni <ntyni@debian.org>
Date: Wed, 17 Sep 2014 21:10:12 +0300
Subject: [PATCH] Test perl packages that include test.pl, not just t/

Both Makefile.PL and Build.PL based build systems will run tests in both
test.pl and t/ if they exist, and so will pkg-perl-autopkgtest as of 0.15.

There are currently 168 affected pkg-perl packages in the Debian sid archive.
---
 lib/testdesc.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/testdesc.py b/lib/testdesc.py
index ea260eb..de55017 100644
--- a/lib/testdesc.py
+++ b/lib/testdesc.py
@@ -312,7 +312,8 @@ def _auto_debian_control_ruby(srcdir, tests):
 def _auto_debian_control_perl(srcdir, tests):
     '''Add automatic test for Perl packages'''
 
-    if (os.path.exists(os.path.join(srcdir, 't')) and
+    if ((os.path.exists(os.path.join(srcdir, 't')) or
+         os.path.exists(os.path.join(srcdir, 'test.pl'))) and
             (os.path.exists(os.path.join(srcdir, 'Makefile.PL')) or
              os.path.exists(os.path.join(srcdir, 'Build.PL')))):
 
-- 
2.1.0

